File: javalin/src/main/java/org/example/joke/model/Joke.java
package org.example.joke.model;

import java.util.UUID;

public record Joke(String id, String setup, String punchline, String category, String source, int lolCount, int groanCount) {
    public static Joke of(String setup, String punchline, String category, String source) {
        return new Joke(UUID.randomUUID().toString(), setup, punchline, category == null ? "" : category, source == null ? "ted" : source, 0, 0);
    }
}

File: swift/Sources/App/Domain/Joke.swift
import Foundation
import Vapor

struct Joke: Codable, Content {
    let id: String
    let setup: String
    let punchline: String
    let category: String
    let source: String
    let lolCount: Int
    let groanCount: Int
    
    enum CodingKeys: String, CodingKey {
        case id
        case setup
        case punchline
        case category
        case source
        case lolCount
        case groanCount
    }
}

struct JokeCreate: Codable, Content {
    let setup: String?
    let punchline: String?
    let category: String?
    let source: String?
    
    enum CodingKeys: String, CodingKey {
        case setup
        case punchline
        case category
        case source
    }
}

struct CountsResponse: Codable, Content {
    let id: String
    let lolCount: Int
    let groanCount: Int
    
    enum CodingKeys: String, CodingKey {
        case id
        case lolCount
        case groanCount
    }
}
